app/models/profile.rb

```diff
@@ -99,31 +99,26 @@ def from_omniauth_hash(omniauth_user_hash)
     self.attributes.merge(update_hash){|key, old, new| old.blank? ? new : old}
   end
 
-  def image_url= url
-    return image_url if url == ''
-    if url.nil? || url.match(/^https?:\/\//)
-      super(url)
-    else
-      super(absolutify_local_url(url))
+  def _set_image_url (att, url)
+    if not url.blank?
+      path = url[/(https?:\/\/(.(?!\/))*.)?(.*)/, 3]
+      if not path.match(/^\//)
+        path += '/'
```

Comment:
The string is created inside the method, modifying it with `<<` instead of creating a new one with `+=` is more efficient.